src/Hacks/autostrafe.cpp

```diff
@@ -9,24 +9,25 @@ void LegitStrafe(C_BasePlayer* localplayer, CUserCmd* cmd)
 	if (localplayer->GetFlags() & FL_ONGROUND)
 		return;
 
-	if (cmd->buttons & IN_FORWARD || cmd->buttons & IN_BACK || cmd->buttons & IN_MOVELEFT || cmd->buttons & IN_MOVERIGHT)
-		return;
-
 	if (cmd->mousedx <= 1 && cmd->mousedx >= -1)
 		return;
 
 	switch (Settings::AutoStrafe::type)
 	{
 		case AutostrafeType::AS_FORWARDS:
-			cmd->sidemove = cmd->mousedx < 0.f ? -450.f : 450.f;
+			cmd->forwardmove=0.f;
+			cmd->sidemove = cmd->mousedx < 0.f ? -450.f :cmd->mousedx==0?0.f: 450.f;
```

Comment:
`cmd->sidemove = cmd->mousedx < 0.f ? -450.f : cmd->mousedx == 0 ? 0.f : 450.f;`

This could also be split into separate if statements to make it more clear what is going on.